haskell-src/Concordium/Crypto/BlsSignature.hs

```diff
@@ -289,6 +291,24 @@ verifyAggregate m pks sig = unsafePerformIO $ do
     withKeyArray ps [] f = withArrayLen ps f
     withKeyArray ps (pk : pks_) f = withPublicKey pk $ \pk' -> withKeyArray (pk' : ps) pks_ f
 
+-- |Verify a signature on the list of bytestrings under the list of public keys.
+-- The public keys are grouped so that the i'th list of public keys corresponds to the
+-- secret keys that signed the i'th message.
+verifyAggregateHybrid :: [ByteString] -> [[PublicKey]] -> Signature -> Bool
```

Comment:
I think it would be cleaner to have a `[(ByteString, [PublicKey])]` parameter, because the lists should be the same length.